### PR TITLE
Support Kubernetes expose property via Quarkus configuration

### LINKED ISSUE #7650: quarkus.{kubernetes,openshift}.expose=true doesn't work
**Describe the bug**

I tried to migrate from the legacy `kubernetes.expose=true` / `openshift.expose=true` properties to the new `quarkus.kubernetes.expose=true` / `quarkus.openshift.expose=true` properties, but no `Ingress` / `Route` is being generated.

Also the `KubernetesConfig` / `OpenshiftConfig` classes don't have an `expose` field, so this looks like an important omission.

**Expected behavior**
Able to generate `Ingress` / `Route` by setting `quarkus.{kubernetes,openshift}.expose=true`, instead of the legacy `{kubernetes,openshift}.expose=true`.

**Actual behavior**
Setting `quarkus.{kubernetes,openshift}.expose=true` doesn't result in generated `Ingress` / `Route`.

**To Reproduce**
1. Add the `quarkus-kubernetes` extension.
2. Run `mvn clean package`.
3. See `target/kubernetes/*.yml`.

**Configuration**
```properties
quarkus.kubernetes.deployment-target=kubernetes,openshift
quarkus.kubernetes.expose=true
quarkus.openshift.expose=true
```

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #7650

### CODE REVIEW

**Comment by geoand** [comment]
I don't have a better solution since I couldn't find a way to create `AddRouteDecorator` and `AddIngressDecorator`.
Unless @iocanel has a trick up his sleeve, I propose we get this in since as @Ladicek mentions this missing configuration is important

**Comment by geoand** [comment]
And now my conscience is clearer since I added a test :)

**Review by iocanel — APPROVED** [review]
LGTM

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java [hunk 1]
```diff
@@ -74,12 +79,28 @@
...
+            Optional<Boolean> unprefixedExpose = config.getOptionalValue(generator + "." + EXPOSE_PROPERTY_NAME,
+                    Boolean.class);
+            Optional<Boolean> prefixedExpose = config
+                    .getOptionalValue(QUARKUS_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, Boolean.class);
+            if ((unprefixedExpose.isPresent() && unprefixedExpose.get())
+                    || (prefixedExpose.isPresent() && prefixedExpose.get())) {
+                unPrefixed.put(DEKORATE_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, true);
+            }
+        }
+    }
```
Comment (machi1990) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java]:
```suggestion
    private static void handleExpose(Config config, Map<String, Object> unPrefixed) {
        for (String generator : EXPOSABLE_GENERATORS) {
            boolean unprefixedExpose = config.getOptionalValue(generator + "." + EXPOSE_PROPERTY_NAME, Boolean.class)
                    .orElse(false);
            boolean prefixedExpose = config.getOptionalValue(QUARKUS_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, Boolean.class)
                    .orElse(false);
            if (unprefixedExpose || prefixedExpose) {
                unPrefixed.put(DEKORATE_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, true);
            }
        }
    }
```

To simply the if condition a bit. WDYT? @geoand

Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java]: Updated, thanks

**Review by machi1990 — APPROVED** [review]
LGTM

**Pull request merged**